lib/onesignal.rb

```diff
@@ -23,6 +23,13 @@ def self.client
     Onesignal::Client.new
   end
 
+  # Exists to allow for realistic testing of initialised configuration
+  # @return [Configuration] The replaced configuration singleton
+  def self.reset_configuration
```

Comment:
I've crated a PR where I extracted the Configuration logic, so it can be tested in isolation: https://github.com/upmysport/onesignal/pull/36 What do you think?